### PR TITLE
Fix google navigation blocking other actions

### LINKED ISSUE #1365: "Open Google" blocks other actions if placed first in a routine
### Prerequisites:
Mic permissions are enabled.
"Allow" or "Don't Allow" button from “Allow Firefox Voice to Collect Voice Transcriptions” pop-up is clicked.
A routine is made that contains an "Open Google", or "Open Google.com" command (this command is placed first), then other "Open" commands. (click mic icon next to the window's menu button, click the "Settings" button from the doorhanger, click "Routines" from the left side of the "Preferences" page, click "+ New routine" button, fill out the fields with the routine name and the actions that need to be done and click the "Save" button)

### STR:
1. Use the shortcut or click on the mic icon from the browser toolbar.
2. Say or write the newly made routine command, e.g. "News".

### Expected result:
New tabs open corresponding to the number of routine actions that need to be made.

### Actual result:
Only the Google tab opens corresponding to the first routine action.

### Notes:
Reproduced on Mac 10.14 and Win10x64 with Firefox Nightly 76.0a1 (2020-03-23).
Not reproduced if the "Open Google"/"Open Google.com" is replaced with a "Search" or "Search Google Images" command in the routine.
Not reproduced  if the "Open Google"/"Open Google.com" is placed last in the routine. (2nd gif)
If the made routine is used multiple times, the other "Open" commands will need changing what websites to open. (see #1351)

### PR DESCRIPTION
**Pull request by dave-ok**
Before submitting a final PR, please:

- [x] Run `npm test` on your machine
- [x] Run `npm run format` to keep code formatted consistently
- [x] Reference the issue you are fixing in at least one of your commit messages, as `Fixes #X`

Fixes #1365
 
@ianb I tried but failed to understand the reason for testing for [google domain](https://github.com/mozilla/firefox-voice/blob/5078ef0840e96e9fe36e2800dbada3f0bf7b1039/extension/background/intentRunner.js#L174) in the `onUpdated` function within the Promise in `createTabGoogleLucky`. Either way it turned out that if a `google.com` url is being loaded the promise never resolves. Introducing an else condition seems to solve this. Am I on the right track?

The utterance `open google` searches for 'open' on google through the `search.searchGoogle` intent instead of using the `navigate.navigate` intent. The matcher in its current state matches phrases like "_Search for cars google_", "_Look for cute kittens google for me_", "_open google_" etc. Modified it to match only when "_on google_" is present. So "_Search for cars on google_" and "_Look for cute kittens on google for me_" will match whereas "_open google_" will not.

Could you please review this? Thanks

### CODE REVIEW

**Comment by ianb** [comment]
@dave-ok I think this is probably not the correct result.

An "I'm lucky" search goes to `https://www.google.com/search?q=query&btnI=` and then often (but not always) goes to `https://www.google.com/url?q=newUrl` and then to the final URL.

I think the issue here is that sometimes the final URL is _also_ a google.com URL, and then it never "finishes" because finishing is defined as leaving google.com.

I think simply improving the `isGoogle` test to match only `/search` might fix this?

**Comment by dave-ok** [comment]
Thanks. That makes things so much clearer. I'll make the changes

**Comment by dave-ok** [comment]
@ianb I have modified the test. Can you please check it?

File: extension/background/intentRunner.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ export class IntentContext {
         ) {
           return;
         }
-        const isGoogle = /^https:\/\/www.google.com\//.test(url);
+        const isGoogle = /\/search/.test(url);
```
Comment (ianb) [inline extension/background/intentRunner.js:174]: I think a somewhat more careful test here might be in order, but I might put it in real quick after landing this

**Comment by dave-ok** [comment]
@ianb I guess I misunderstood the point you were trying to pass across. Thank you very much

**Pull request merged**